feat(ai): support multiple AI providers (#2)
rafaelrcamargo/shipit: README.md

[^1]: Self-criticism from the creator here, I made this tool because my own Git history was a disaster. :D

> [!IMPORTANT]
> This tool requires one of the following AI provider API keys to be set as an environment variable:
>
> - **Claude (Anthropic)**: `ANTHROPIC_API_KEY` - Get yours at <https://console.anthropic.com/>
> - **GPT (OpenAI)**: `OPENAI_API_KEY` - Get yours at <https://platform.openai.com/api-keys>
> - **Gemini (Google)**: `GOOGLE_GENERATIVE_AI_API_KEY` - Get yours at <https://aistudio.google.com/app/apikey>
>
> The tool will automatically detect and use the first available API key in the order above. The free tiers should be sufficient for running this tool.

## Setup

```bash
# Build it and copy to /usr/local/bin/shipit
bun run build && sudo bun run copy

> # Set one of these API keys (tool will auto-detect which one to use):
> 
> # For Claude (Anthropic) - Recommended
> export ANTHROPIC_API_KEY="sk-ant-..."
> 
> # For GPT (OpenAI)
> export OPENAI_API_KEY="sk-..."
> 
> # For Gemini (Google) - Fallback option
> export GOOGLE_GENERATIVE_AI_API_KEY="..."

Reviewer: ```suggestion
# Set one of these API keys (the tool will automatically detect which one to use):

export ANTHROPIC_API_KEY="..."
export OPENAI_API_KEY="..."
export GOOGLE_GENERATIVE_AI_API_KEY="..."
```
